components/elm/src/biogeochem/EcosystemDynMod.F90

```diff
@@ -144,115 +144,106 @@ subroutine EcosystemDynLeaching(bounds, num_soilc, filter_soilc, &
     integer                  , intent(in)    :: filter_pcropp(:)  ! filter for prognostic crop patches
     logical                  , intent(in)    :: doalb             ! true = surface albedo calculation time step
     type(cnstate_type)       , intent(inout) :: cnstate_vars
-    type(carbonflux_type)    , intent(inout) :: carbonflux_vars
-    type(carbonstate_type)   , intent(inout) :: carbonstate_vars
-    type(carbonflux_type)    , intent(inout) :: c13_carbonflux_vars
-    type(carbonstate_type)   , intent(inout) :: c13_carbonstate_vars
-    type(carbonflux_type)    , intent(inout) :: c14_carbonflux_vars
-    type(carbonstate_type)   , intent(inout) :: c14_carbonstate_vars
-    type(nitrogenflux_type)  , intent(inout) :: nitrogenflux_vars
-    type(nitrogenstate_type) , intent(inout) :: nitrogenstate_vars
-    type(waterstate_type)    , intent(in)    :: waterstate_vars
-    type(waterflux_type)     , intent(in)    :: waterflux_vars
     type(frictionvel_type)   , intent(in)    :: frictionvel_vars
     type(canopystate_type)   , intent(inout) :: canopystate_vars
 
-    type(phosphorusflux_type)  , intent(inout) :: phosphorusflux_vars
-    type(phosphorusstate_type) , intent(inout) :: phosphorusstate_vars
-
+    character(len=64) :: event
+    real(r8) :: dt
     !-----------------------------------------------------------------------
-  
-    !if(.not.(use_pflotran.and.pf_cmode)) then
-    call t_startf('PhosphorusWeathering')
+    dt = dtime_mod;
+
+    ! only do if ed is off
+    event = 'PhosphorusWeathering'
+    call t_startGPU(event)
     call PhosphorusWeathering(num_soilc, filter_soilc, &
-         cnstate_vars,phosphorusstate_vars,phosphorusflux_vars)
-    call t_stopf('PhosphorusWeathering')
-    
-    call t_startf('PhosphorusAdsportion')
+         cnstate_vars, dt)
+    call t_stopGPU(event)
+
+    event = 'PhosphorusAdsportion'
+    call t_startGPU(event)
     call PhosphorusAdsportion(num_soilc, filter_soilc, &
-         cnstate_vars,phosphorusstate_vars,phosphorusflux_vars)
-    call t_stopf('PhosphorusAdsportion')
-    
-    call t_startf('PhosphorusDesoprtion')
+         cnstate_vars, dt)
+    call t_stopGPU(event)
```

Comment:
t_startGPU/t_stopGPU(event) are easy to reuse. However, it's unclear what scope they refer to. So, perhaps include the 'timer_name' in the call invocations. I guess this would probably reveal a missing t_startf/t_stopf that's causing issues.

Reply:
It looks like you removed the timers on the CPU side. Would timing info be absent when someone runs this just on the CPU?

Reply:
@peterdschwartz Revisiting this topic of timers: It's still unclear how t_stopGPU is implemented and how it handles CPU only runs. 

Are these very fine-grained timers for GPU? If so, you are probably better off with nvprof/nsys style tool. The GPTL timers are primarily meant for application phases.